types/state/stateTxn.go

```diff
@@ -0,0 +1,78 @@
+package state
+
+import (
+	"errors"
+	"fmt"
+	"github.com/AccumulateNetwork/accumulate/smt/storage"
+	"github.com/AccumulateNetwork/accumulate/smt/storage/database"
+	"github.com/AccumulateNetwork/accumulate/types"
+)
+
+type DBTransaction struct {
+	state        *StateDB
+	updates      map[types.Bytes32]*blockUpdates
+	writes       map[storage.Key][]byte
+	transactions transactionLists
+}
+
+func (tx *DBTransaction) WriteStates(blockHeight int64) ([]byte, int, error) {
+	panic("implement me")
+}
```

Comment:
Is this needed? My intent was to replace `StateDB.WriteStates` with `DBTransaction.Commit`, which you've done. So I think this can be deleted.

Reply:
this method is required to make DBTransaction an implementation of DB interface otherwise it shows error.